otherlibs/stdune/src/fpath.ml

```diff
@@ -1,4 +1,10 @@
-let is_root t = Filename.dirname t = t
+let is_root =
+  if Sys.unix
```

Comment:
I wonder: don't we have something better for OS deterction in Stdune?

Reply:
Oh, yes, it seems we do: `Platform.OS.value`.

Reply:
Why? It seems nicer, especially given that some future implementations might only, say, work on Windows but not on other platforms.

I've started internal benchmarks with the following version:

```
let is_root =
  match Platform.OS.value with
  | Linux | Darwin -> fun x -> x = "/" || x = "."
  | Windows | Other ->
    (* CR-someday rgrinberg: Can we do better on Windows? *)
    fun s -> Filename.dirname s = s
;;
```

Reply:
`Other` are almost always Unix-like so they should be included for the optimization. This is why `Sys.unix` is a bit more precise, it's true for every OS with `/` at the root.

Reply:
Hmm, I see. That makes sense though I don't like mixing different styles (i.e. sometimes using `Sys`, sometimes using `Platform`). Maybe `Platform` should expose `slash_is_root : bool` or something like that.

Reply:
The rest of the module is already using Sys so perhaps we should just it all at once to use Platform (along with introducing this function)

Reply:
OK, for now, I've switched to `Sys.unix` internally as well, but yes, it seems useful to do a separate clean up in future.